node/pkg/telemetry/prom_remote_write/scrape.go

```diff
@@ -0,0 +1,84 @@
+package promremotew
+
+import (
+	"bytes"
+	"context"
+	"net/http"
+
+	"github.com/golang/snappy"
+	"github.com/prometheus/client_golang/prometheus"
+	dto "github.com/prometheus/client_model/go"
+	"go.uber.org/zap"
+	"google.golang.org/protobuf/proto"
+)
+
+type PromTelemetryInfo struct {
+	PromRemoteURL string
+	NodeName      string
+}
+
+func scrapeLocalMetricsViaGatherer() (map[string]*dto.MetricFamily, error) {
+	metrics, err := prometheus.DefaultGatherer.Gather()
+	if err != nil {
+		return nil, err
+	}
+	// Here we have an array of metrics.
+	mapByName := map[string]*dto.MetricFamily{}
+	for i := 0; i < len(metrics); i++ {
+		met := metrics[i]
+		name := met.GetName()
+		if _, ok := mapByName[name]; !ok {
+			mapByName[name] = met
+		}
+	}
+	return mapByName, nil
+}
+
+func ScrapeAndSendLocalMetrics(ctx context.Context, info PromTelemetryInfo, logger *zap.Logger) error {
+	metrics, err := scrapeLocalMetricsViaGatherer()
+	if err != nil {
+		logger.Error("Could not scrape local metrics", zap.Error(err))
+		return err
+	}
+	labels := map[string]string{"node_name": info.NodeName}
```

Comment:
Maybe add labels for the following (like in logging):

- Product
- Network
- Version

Reply:
Done